src/hooks/transactions/usePayload.ts

```diff
@@ -0,0 +1,67 @@
+// Copyright 2021 Parity Technologies (UK) Ltd.
+// This file is part of Parity Bridges UI.
+//
+// Parity Bridges UI is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Parity Bridges UI is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Parity Bridges UI.  If not, see <http://www.gnu.org/licenses/>.
+
+import { compactAddLength } from '@polkadot/util';
+import { useAccountContext } from '../../contexts/AccountContextProvider';
+import { useCallback } from 'react';
+import { useApiCallsContext } from '../../contexts/ApiCallsContextProvider';
+import { useSourceTarget } from '../../contexts/SourceTargetContextProvider';
+import { TransactionActionCreators } from '../../actions/transactionActions';
+import { useUpdateTransactionContext } from '../../contexts/TransactionContext';
+import logger from '../../util/logger';
+
+export const usePayload = () => {
+  const { createType } = useApiCallsContext();
+  const {
+    sourceChainDetails: { chain: sourceChain }
+  } = useSourceTarget();
+  const { dispatchTransaction } = useUpdateTransactionContext();
+  const { account } = useAccountContext();
+
+  const getPayload = useCallback(
```

Comment:
This function should not be called `getPayload`, because it's not returning anything. It should be called `updatePayload`, but IMHO this should even be a hook at all - it would be much easier to make it a regular helper funciton that looks like this:
```
function getPayload(createType, chain, account) {
 ...
 return payload
}

/// Usage
const { createType } = useApiCallsContext();
const { account } = useAccountContext();
const {
    sourceChainDetails: { chain: sourceChain }
 } = useSourceTarget();
try {
  const payload = getPayload(createType, sourceChain, account);
  dispatchTransaction(TransactionActionCreators.setPayload(payload));
} catch (e) {
  dispatchTransaction(TransactionActionCreators.setTransactionError(error, 'getPayload'));
}
```
The more we extract to regular non-React dependent functions the easier the code is to understand. You separate library logic from your domain-logic of the application.
IMHO this shouldn't really even be done by a component/hook - this is part of the data model really. Every time `call` changes the `payload` should be constructed right away.
Like these two things are tightly coupled, so we should be always updating `{call, payload}` together and never first `call` and then count on some hooks being present to auto-update the `payload`. It can either be done inside a reducer (but yeah, with current design you will run into issues of: "I need all of these parameters that are currently hooks") or simply on the call site that updates `call` in local/global state.

Some other issues I've noticed with the code:
- `createType` is something coming from polkadot.js/api - how does it correlate to `sourceChain`?
- It seems to me that this code would simply be broken if we use `createType` from `useApiCallsContext` and `targetChain`.
- This indicates some issue with API design - why these two things are separated? How do I get `createType` of the other chain?